js/adapt-xapi.js

```diff
@@ -93,10 +105,45 @@ define(function(require) {
         return;
       }
 
-      var statement = statementModel.getStatementObject();
+      this.sendStatement(
+        statementModel.getStatement()
+      );
+    },
+
+    onQuestionAttempt: function(question) {
+      if (!question.model.get('_recordInteraction')) {
+        return;
+      }
+
+      var data = {
+        activityId: this.get('activityId'),
+        actor: this.get('actor'),
+        registration: this.get('registration'),
+        model: question.model
+      };
+
+      var statementModel;
+      switch (question.model.get('_component')) {
+        case "mcq":
```

Comment:
In that case though each specific question component would need to decorate the question model in its own way, so doesn't that mean the tight coupling is unavoidable?